ansible/library/interface_facts.py

```diff
@@ -21,6 +21,44 @@
 from ansible.module_utils.basic import *
 from collections import defaultdict
 
+DOCUMENTATION = '''
+---
+module: interfaces_facts
+version_added: "1.1"
+author: John Arnold
+short_description: Retrive interface facts from device
+description:
+    - Retrieve interface facts for a device, the facts will be
+      inserted to the ansible_facts key.
+options:
+    all_ports:
+        description:
+            - all ports that expected to be at link up status
+            - this parameter doens't limit how many ports can
+              be present on the device, interface_facts returns
+              facts of all existing ports.
+        required: false
+    except_port:
+        description:
+            - an exeption of port from all_ports list. The link
+              status of this port is not checked.
+'''
+
+EXAMPLES = '''
+# Gather minigraph facts
+- name: Gathering minigraph facts about the device
+  interface_facts:
+
+# Gather minigraph facts and check a set of ports should be up
+- name: Gathering minigraph facts about the device
+  interface_facts: all_ports={{minigraph_ports}}
+
+# Gather minigraph facts and check a set of ports should be up except one
+- name: Gathering minigraph facts about the device
+  interface_facts: all_ports={{minigraph_ports}} except_port={{except_port}}
```

Comment:
>except_port [](start = 49, length = 11)

ignore_ports

Reply:
This parameter has been removed.

Thanks,
Ying